### ISSUE TITLE
Migrating to 3.4 erased all my Pinned folders !!!

### ISSUE THREAD

**Issue by MarathonMax** [files-community/Files#15410]
### Description

Migrating to 3.4 erased all my Pinned folders !!!

### Steps To Reproduce

Migrating to 3.4 erased all my Pinned folders !!!

### Requirements

Migrating to 3.4 erased all my Pinned folders !!!

### Files Version

3.4.0.0

### Windows Version

10.0.22631.3593

### Log File

[debug.log](https://github.com/files-community/Files/files/15337244/debug.log)

**Comment by Josh65-2201** [comment]
Thanks for the report, 
- Please see if they are still shown in Windows File Explorer
- Try clearing recent folders in Windows File Explorer settings > `...` > Options > Clear and disabling show recent folders in > that same menu.

**Comment by MarathonMax** [comment]
I do not see this option - I have checked them all I believe. Then again my
software is in French

Maxime

**Comment by Josh65-2201** [comment]
For what step aren't you seeing the option?

Windows File Explorer options here
![image](https://github.com/files-community/Files/assets/26977075/b9056086-dfd9-473b-9695-c3d73ff49b7e)


Files option here
![image](https://github.com/files-community/Files/assets/26977075/82e27092-d1a9-4592-b83b-953b6ae713fe)

**Comment by MarathonMax** [comment]
These that you asked for

Thanks for the report,

   - Please see if they are still shown in Windows File Explorer
   - Try clearing recent folders in Windows File Explorer settings > ... >
   Options > Clear and disabling show recent folders in > that same menu.


Maxime

**Comment by Josh65-2201** [comment]
Do the screenshots I send help?
Also do you have Files settings > Advanced > Open Files on Windows startup enabled?

**Comment by MarathonMax** [comment]
Second screen (black with green handwritten circle): my option is set to ACTIVATED

As for clearing File Explorer history ... I do not see where that is.

**Comment by Josh65-2201** [comment]
Please try closing Files from the system tray and opening from the start menu.

**Comment by MarathonMax** [comment]
OK got it where to clear history. Just did it. Relaunched Files (your app) and problem is fixed. Tks.

**Comment by Josh65-2201** [comment]
Thanks for the information do you also have the leave running in background option on?

**Comment by MarathonMax** [comment]
yes I leave that ON but turned it off after clearing the Windows FIle
Explorer list just to ensure that Files would boot clean. I did turn it
back on afterwards.

Maxime

**Comment by JudeLC** [comment]
Exact same issue for me.

**Comment by MarathonMax** [comment]
The problem came. I redid the szme steps and was able to get back my pinned
folders. Obviously the bug is still present :-(

Maxime

**Comment by yaira2** [comment]
Can you try the solutions mentioned here and let us know if they helped?
https://answers.microsoft.com/en-us/windows/forum/all/windows-11-quick-access-folders-gone-and-pin-to/49b3f66a-062f-43e6-8417-728e2a42c896

**Comment by yaira2** [comment]
@hishitetsu do we reload the pinned items when resuming from the background?

**Comment by hishitetsu** [comment]
No, but if the app starts in the background when Windows starts, they will be loaded the first time the app is opened.

**Comment by yaira2** [comment]
I'm not sure why it's sometimes clearing the list, but would adding logic to check if they need to be reloaded make sense?

**Comment by hishitetsu** [comment]
It is not being cleared, but rather it is failing to load from the beginning, I guess.

**Comment by schwupi** [comment]
If you change the UI language in Files the pinned items reappear.

**Comment by mrlopezco** [comment]
> If you change the UI language in Files the pinned items reappear.

Interesting, this also fixed it for me. I changed it from "Windows default" to "English (USA)" and my pinned items are back now.

**Comment by hishitetsu** [comment]
> > If you change the UI language in Files the pinned items reappear.
> 
> Interesting, this also fixed it for me. I changed it from "Windows default" to "English (USA)" and my pinned items are back now.

That is because the app is restarted when changing the language.

**Comment by theyv** [comment]
Same issue here. Clearing recents fixed the thing for now.

**Comment by TheCiROMG** [comment]
> Please try closing Files from the system tray and opening from the start menu.

Closing it from the taskbar, and opening it from startup works and shows everything correctly, if it is fixed in the taskbar or starts with the system, it does not show the pinned folders and sometimes the double view is bugged (the bug is basically that it duplicates the disks and folders).

**Comment by tommy-IKIIKI** [comment]
I know it is not a "fix".
but if you add a New Window, it can reload the pins without restarting the whole app.

**Comment by jamesmusik** [comment]
Pinning a new item also reloads the pins.

**Comment by verdaderoken** [comment]
> I know it is not a "fix".
> but if you add a New Window, it can reload the pins without restarting the whole app.

This works! Hope this gets fixed

**Comment by 0x5bfa** [comment]
Working on this in #16485

**Comment by ndekross** [comment]
I am also experiencing this problem.

**Comment by apoorvpotnis** [comment]
I am experiencing the same issue. And after closing and reopening, with leaving the `Leave app running in the background` open, I am unable to pin any folders to the sidebar. Version. 3.8.0.0

**Comment by gokhanayisigi** [comment]
Same issue, after a while (like after 40 mins to run in background), suddenly pinned items not showing in app.

**Comment by foinf** [comment]
Ssame issue. Works for a while but then pinned items disappear. Using Task Manager to kill Files and then reopening it fixes it for a while.

**Comment by foinf** [comment]
> [@foinf](https://github.com/foinf) Follow the work around steps at the top of this issue to fix it

Hi, 
I am aware of the workaround. I meant to report that it doesn't work if I let it run with the stay open in the background option.

Thanks for getting back to me.

**Comment by 0x5bfa** [comment]
> Working on this in #16485.
> 
> Ridiculously the refresh button also doesn't work and so, once we finished working on both this and the simplification of IStorageDevicesService, we're gonna work on making the refresh button work.

@yaira2 At this point, I'm working on Omnibar but I already opened a PR for this and it got closed. Do you think we should reopen/rework this for real quick and review to make it to v4?

**Comment by gokhanayisigi** [comment]
is there any update about this issue? It's so annoying that I cannot reach my pinned items.

**Comment by Josh65-2201** [comment]
See the steps at the top of the page

**Comment by gokhanayisigi** [comment]
Cleared recents, not solved my issue. Only unchecking "Leave app running in back" option solves but it makes that app launch is much much slower. @Josh65-2201

**Comment by Josh65-2201** [comment]
That's expected as the settings off, you can minimize Files rather than closing and you won't have the issue of pins not loading.

**Comment by zurce** [comment]
This is probably the biggest issue keeping me for using files, it's so frequent it's incredible it hasn't been caught up

For reference once it happens switching the languages of the app makes them come back, so i'm frequently changing between English UK and US

**Comment by Josh65-2201** [comment]
See the workaround at the top of the page

**Comment by yair100** [comment]
> This is probably the biggest issue keeping me for using files, it's so frequent it's incredible it hasn't been caught up
> 
> For reference once it happens switching the languages of the app makes them come back, so i'm frequently changing between English UK and US

We believe the issue was resolved for the Home Page in v4 (please let us know if this is not the case). We're working to do the same for the sidebar, but it's a little more challenging due to the coding structure there.

**Comment by HyPeRXKo** [comment]
Still an issue on 4.0.11.0, it seems to do it randomly, a full restart of the app (closing the background app than reopening) seems to fix it temporarily (without any of the workaround mentioned here applied).

**Comment by domsarc** [comment]
Have same issue and no solution really works so I am unfortunately still bound to use default file manager. I hope this gets fixed in the future because I would prefer to use Files.


Edit: Also for some reason my Recylce bin doesnt work after restarting the computer. The only solution is that I need to close and start Files again.

<img width="571" height="565" alt="Image" src="https://github.com/user-attachments/assets/6e5b9233-9c39-4af2-98d5-6fa82ca76ce2" />

**Comment by Josh65-2201** [comment]
Did you use the workaround at the top of the issue?

**Comment by domsarc** [comment]
> Did you use the workaround at the top of the issue?

Yes and also I have the Show recommended section unchecked by default from before I installed Files.

**Comment by wreathe-unreal** [comment]
Still an issue on v4 but thank you for the workaround.

**Comment by yair100** [comment]
Is anyone able to reproduce this in v4.1.3?

**Comment by yair100** [comment]
This appears to be resolved.
